packages/calendar/components/ChannelFilter/index.jsx

```diff
@@ -25,14 +32,35 @@ const ChannelFilter = ({ channelFilter, channelSelected }) => {
   }`;
   const label = allChannelsAreSelected ? 'All channels' : labelForSoloChannels;
 
+  const handleFilteredChannels = ({ beforeFilterCount, afterFilterCount }) => {
+    trackChannelsFiltered({
+      isWeekly,
+      beforeFilterCount,
+      afterFilterCount,
+      isViewOnly,
+    });
+  };
+
+  const handleSelectedChannels = ({ beforeFilterCount, afterFilterCount }) => {
+    trackChannelsSelected({ isWeekly, beforeFilterCount, afterFilterCount });
+  };
+
   // Select default filter values
   useEffect(() => {
     // Filter by one channel if coming from route with channel state (e.g. queue)
     if (channelSelected) {
       setChannels([channelSelected]);
+      handleFilteredChannels({
+        beforeFilterCount: channels.length,
```

Comment:
Since when the page loads this is the first code to run, this can be replaced:
```suggestion
        beforeFilterCount: 0,
```